qiskit_nature/runtime/vqe_client.py

```diff
@@ -299,7 +305,9 @@ def compute_minimum_eigenvalue(
         vqe_result.cost_function_evals = result.get("cost_function_evals", None)
         vqe_result.eigenstate = result.get("eigenstate", None)
         vqe_result.eigenvalue = result.get("eigenvalue", None)
-        vqe_result.aux_operator_eigenvalues = result.get("aux_operator_eigenvalues", None)
+        vqe_result.aux_operator_eigenvalues = dict(
+            zip(aux_operators, result.get("aux_operator_eigenvalues", None))
+        )
```

Comment:
I would expect that the conversion to `dict` is depended on the type of the `aux_operators` passed into this method and the value of `qiskit_nature.settings.dict_aux_operators` (where the latter should take precedence (and overwrite) the former).

Reply:
Since this implements the MinimumEigenSolver interface from Qiskit Terra, I think ideally it ought to follow that, which means List in -> List out, and Dict in - > Dict out as far as aux ops and their result goes. While this class does sit in Nature, its really a runtime version of VQE and I think it ought to behave as the same way. Since Nature will create Dicts it should see only Dicts back even if the runtime here supports List.

Reply:
Good points; I completely agree with the 'Type in -> Type out' approach.
Internally in VQEClient I think it does not matter that much, so let's stick to the converted Dicts (also using them as input for the runtime even if the input to VQEClient was a List).
I see your point @mrossinek about `qiskit_nature.settings.dict_aux_operators` but I think it's better to ignore this setting since the convertion to dict can't restore the correct names for the aux_operators but only fill it with numbers counting up. So, it would end up claiming to correctly convert the List to Dict but in fact doing only filling up with dummy keys. And with `qiskit_nature.settings.dict_aux_operators` set to True it should get a Dict as input either way as @woodsp-ibm pointed out -- else things would be broken already at another place.

Reply:
Yes you have convinced me that the `Type in -> Type out` approach is the way to go here :+1: 
This probably also resolves our conversation above, right @cpossel ?

Reply:
Yes all suggestions from the two of you are included in the newest commit. I'll leave it to you to 'resolve conversation' if you are fine with the changes made.